app/models/service_template_tenant.rb

```diff
@@ -0,0 +1,4 @@
+class ServiceTemplateTenant < ApplicationRecord
+  belongs_to :service_template, :dependent => :destroy
```

Comment:
@gtanzillo I had it vice versa.

so I have  `:dependency => :destroy`  on [ServiceTemplate#additional_tenants](https://github.com/ManageIQ/manageiq/pull/18554/commits/79352578bc76f16b35472a49b9848f1c83d57c8f#diff-816ef92b3b154ddd9cf6c420b719ba26R41) (same for [Tenant#service_templates) which means that it will destroy join table `ServiceTemplateTenant` and it is GOOD when ServiceTemplate is destroyed (or Tenant)

also I remove  `:dependent => :destroy` from `belongs_to` from ServiceTemplateTenant and it means that it will not destroy `Tenant` when you destroy `ServiceTemplate`

so `ServiceTemplate.first.destroy` will destroy only `ServiceTemplateTenant` records and NOT Tenants.